src/main/java/uk/gov/companieshouse/web/accounts/service/cic/impl/CicApprovalServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+import uk.gov.companieshouse.web.accounts.service.cic.CicReportService;
+import uk.gov.companieshouse.web.accounts.transformer.cic.CicApprovalTransformer;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+import uk.gov.companieshouse.web.accounts.validation.helper.ServiceExceptionHandler;
+
+@Service
+public class CicApprovalServiceImpl implements CicApprovalService {
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private ServiceExceptionHandler serviceExceptionHandler;
+
+    @Autowired
+    private CicApprovalTransformer transformer;
+
+    @Autowired
+    private CicReportService cicReportService;
+
+    private static final UriTemplate APPROVAL_URI =
+        new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/cic-report/cic-approval");
+
+    private static final String DAY_MONTH_REGEX = "\\d{1,2}";
+
+    private static final String YEAR_REGEX = "\\d{4}";
+
+    private static final String APPROVAL_DATE_FIELD_PATH = "date";
+
+    private static final String APPROVAL_DATE_ERROR_LOCATION = ".approval.date";
+
+    private static final String DATE_MISSING =
+        "validation.date.missing" + APPROVAL_DATE_ERROR_LOCATION;
+
+    private static final String DATE_INCOMPLETE =
+        "validation.date.incomplete" + APPROVAL_DATE_ERROR_LOCATION;
+
+    private static final String DATE_FORMAT_INVALID =
+        "validation.date.format" + APPROVAL_DATE_ERROR_LOCATION;
+
+    private static final String DATE_INVALID = "validation.date.nonExistent";
+
+    private static final String RESOURCE_NAME = "approval";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<ValidationError> submitCicApproval(String transactionId, String companyAccountsId,
+        CicApproval cicApproval) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = APPROVAL_URI.expand(transactionId, companyAccountsId).toString();
+
+        CicApprovalApi cicApprovalApi = transformer.getCicApprovalApi(cicApproval);
+
+        try {
+            CicReportApi cicReportApi = cicReportService.getCicReport(apiClient, transactionId,
+                companyAccountsId);
+
+            if (cicReportApi.getLinks().getApproval() != null) {
+                apiClient.cicReport().approval().update(uri, cicApprovalApi).execute();
+            } else {
+                apiClient.cicReport().approval().create(uri, cicApprovalApi).execute();
+            }
+        } catch (ApiErrorResponseException e) {
+            return serviceExceptionHandler.handleSubmissionException(e, RESOURCE_NAME);
+        } catch (URIValidationException e) {
+            serviceExceptionHandler.handleURIValidationException(e, RESOURCE_NAME);
+        }
+
+        return new ArrayList<>();
+    }
+
+    @Override
+    public List<ValidationError> validateCicApprovalDate(CicApproval cicApproval) {
```

Comment:
All of this logic already exists in the `ApprovalServiceImpl` class, we should pull this into a generic validator class